Vagrantfile

```diff
@@ -0,0 +1,81 @@
+# Load configuration for the-vagrant
+require_relative 'src/TheVagrant.rb'
+the_vagrant = TheVagrant.new($the_vagrant_project_dir, $the_vagrant_config)
+
+
+# Ensure Vagrant requirements are present
+Vagrant.require_version ">= 2.1.0"
+
+%w{ vagrant-hostmanager vagrant-auto_network }.each do |plugin|
+    unless Vagrant.has_plugin?(plugin)
+        raise "#{plugin} plugin is not installed. Please install with: vagrant plugin install #{plugin}"
+    end
+end
+
+
+# Vagrant setup
+Vagrant.configure(2) do |config|
+
+    config.hostmanager.enabled = true
+    config.hostmanager.manage_host = true
+
+    config.vm.define "#{the_vagrant.config['project']}" do |box|
+
+        box.vm.box = "palantir/drupalbox"
+        box.vm.box_version = ">= 1.2.0, < 2.0"
+
+        box.vm.provider "virtualbox" do |vb|
+            vb.customize ["modifyvm", :id, "--memory", "2048", "--audio", "none"]
+        end
+
+        box.vm.hostname = "#{the_vagrant.config['hostname']}"
+        box.vm.network :private_network, :auto_network => true
+
+        box.hostmanager.aliases = the_vagrant.config['extra_hostnames']
+
+        box.vm.synced_folder ".", "/vagrant", :disabled => true
+        box.vm.synced_folder ".", "/var/www/#{the_vagrant.config['hostname']}", :nfs => true
+
+        box.ssh.forward_agent = true
+    end
+
+    config.vm.provision "the-vagrant", type: "ansible" do |ansible|
+        ansible.playbook = the_vagrant.config['playbook']
+
+        ansible.groups = {
+            "all:children" => ["#{the_vagrant.config['project']}"]
+        }
+
+        ansible.extra_vars = {
+            "project" => the_vagrant.config['project'],
```

Comment:
@becw, now that these are already in an array, couldn't we just pass them as `ansible.extra_vars = the_vagrant.config`? I haven't finished reviewing everything, so maybe there's a reason this wouldn't work that I'm not thinking of yet.

Reply:
I do see that some of the keys aren't the same, but could we rename one or the other?

Reply:
I think it could make sense to update some of the keys... I was also wondering about mapping the key names? Basically some ansible roles use variables, and anything we pass in here gets directly passed through to the ansible roles... which can come in handy when you know about it (e.g. for a few PHP settings).